components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java

```diff
@@ -114,46 +113,37 @@ public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRe
                 registrationRequestProfile = new RegistrationRequestProfile();
             }
 
-            Object obj = jsonData.get(RegistrationRequest.RegisterRequestConstant.REDIRECT_URIS);
+            Object obj = jsonData.get(RegistrationRequest.RegisterRequestConstant.GRANT_TYPES);
             if (obj != null && obj instanceof JSONArray) {
-                JSONArray redirectUris = (JSONArray) obj;
-                for (Object redirectUri : redirectUris) {
-                    //ToDO invalid_redirect_uri - error code should be sent to client in below error cases.
-
-                    if (redirectUri == null || IdentityUtil.isBlank(redirectUri.toString())) {
-                        throw IdentityException
-                                .error(FrameworkClientException.class, "The redirection URI is either null or blank.");
+                JSONArray grantTypes = (JSONArray) obj;
+                for (Object grantType : grantTypes) {
+                    if (IdentityUtil.isNotBlank((String) grantType)) {
```

Comment:
Better to check the type before casting, and use the casted, or converted variable at line 121 too.